src/main/java/ch/njol/skript/conditions/CondIsWhitelisted.java

```diff
@@ -19,72 +19,72 @@
 package ch.njol.skript.conditions;
 
 import org.bukkit.Bukkit;
-import org.bukkit.entity.Player;
+import org.bukkit.OfflinePlayer;
 import org.bukkit.event.Event;
 import org.eclipse.jdt.annotation.Nullable;
 
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Condition;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
 @Name("Is Whitelisted")
-@Description("Whether or not the server or a player is whitelisted.")
-@Examples({"if server is whitelisted:", "if player is whitelisted"})
-@Since("2.5.2")
+@Description("Whether or not the server or a player is whitelisted, and server is whitelist enforced.")
+@Examples({
+	"if player is whitelisted:",
+	"if server is whitelisted:",
+	"if server is whitelist enforced:"
+})
+@Since("2.5.2, INSERT VERSION (enforce)")
+@RequiredPlugins("Minecraft 1.17+")
 public class CondIsWhitelisted extends Condition {
-	
+
+	private static final boolean ENFORCE_SUPPORT;
+
 	static {
+		ENFORCE_SUPPORT = Skript.methodExists(Bukkit.class, "isWhitelistEnforced");
 		Skript.registerCondition(CondIsWhitelisted.class,
 			"[the] server (is|1¦is(n't| not)) white[ ]listed",
-			"%players% (is|are)(|1¦(n't| not)) white[ ]listed");
+			"%offlineplayers% (is|are)(|1¦(n't| not)) white[ ]listed",
+			(ENFORCE_SUPPORT ? "[the] server (is|1¦is(n't| not)) white[ ]list enforced" : ""));
 	}
-	
+
 	@Nullable
-	private Expression<Player> player;
-	
+	private Expression<OfflinePlayer> player;
+
 	private boolean isServer;
-	
+	private boolean isEnforce;
+
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setNegated(parseResult.mark == 1);
-		isServer = matchedPattern == 0;
-		if (matchedPattern == 1)
-			player = (Expression<Player>) exprs[0];
+		if (matchedPattern == 0 || matchedPattern == 2) {
+			isServer = true;
+			isEnforce = matchedPattern == 2;
+		}
+		else if (matchedPattern == 1)
+			player = (Expression<OfflinePlayer>) exprs[0];
```

Comment:
```suggestion
		isServer = matchedPattern == 0;
		isEnforce = matchedPattern == 2;
		if (matchedPattern == 1)
			player = (Expression<OfflinePlayer>) exprs[0];
```
I would prefer to see this part simplified :)